src/theory/sets/cardinality_extension.cpp

```diff
@@ -57,20 +65,105 @@ void CardinalityExtension::registerTerm(Node n)
     d_eqc_to_card_term[r] = n;
     registerCardinalityTerm(n[0]);
   }
-  if (tnc.isInterpretedFinite())
+  Trace("sets-card-debug") << "...finished register term" << std::endl;
+}
+
+void CardinalityExtension::checkFiniteTypes()
+{
+  for(std::pair<const TypeNode, bool> & pair: d_t_card_enabled)
   {
-    std::stringstream ss;
-    ss << "ERROR: cannot use cardinality on sets with finite element "
-          "type (term is "
-       << n << ")." << std::endl;
-    throw LogicException(ss.str());
-    // TODO (#1123): extend approach for this case
+    TypeNode type = pair.first;
+    if(pair.second && type.isInterpretedFinite())
+    {
+      checkFiniteType(type);
+    }
+  }
+}
+
+void CardinalityExtension::checkFiniteType(TypeNode & t)
+{
+  Assert(t.isInterpretedFinite());
+  NodeManager* nm = NodeManager::currentNM();
+  // get the universe set of type t
+  // ToDo: investigate refactoring nm->mkSetType
```

Comment:
TODOs should refer to an issue (if the issue doesn't exist yet, please create one).

Reply:
Actually I don't think it needs refactoring any more. It was weird to me that I need to get a `(Set t)` out of type `t` to get the universe set. But now I think it is more consistent with SMT-LIB  `(as univset (Set t))`. So I removed this `ToDo` comment